Analyses/src/BFieldPlotter_module.cc

```diff
@@ -118,34 +130,46 @@ namespace mu2e {
 
   //fill a histogram with the magnetic field
   void BFieldPlotter::fillHistogram(BFMap const *map, art::ServiceHandle<art::TFileService> tfs) {
+    GeomHandle<DetectorSystem> det;
     //define histogram binning such that map edges are bin centers and inside histogram
     long nbinsOne = std::lround((_axisOneMax - _axisOneMin)/_mapBinSize) + 1;
     long nbinsTwo = std::lround((_axisTwoMax - _axisTwoMin)/_mapBinSize) + 1;
     //check that the map bin step size works with the edges given
-    if(std::abs(_axisOneMin + (nbinsOne-1)*_mapBinSize - _axisOneMax > (_axisOneMax-_axisOneMin)/(100.*(nbinsOne-1))))
+    if(nbinsOne > 1 && std::abs(_axisOneMin + (nbinsOne-1)*_mapBinSize - _axisOneMax > (_axisOneMax-_axisOneMin)/(100.*(nbinsOne-1))))
       throw cet::exception("BADCONFIG") << "BField mapping axis values not steppable with step size given: "
 					<< _axisOneMin << " < axisOneValues < " << _axisOneMax << ", "
 					<< "step size = " << _mapBinSize;
-	
-    if(std::abs(_axisTwoMin + (nbinsTwo-1)*_mapBinSize - _axisTwoMax > (_axisTwoMax-_axisTwoMin)/(100.*(nbinsTwo-1))))
+
+    if(nbinsTwo > 1 && std::abs(_axisTwoMin + (nbinsTwo-1)*_mapBinSize - _axisTwoMax > (_axisTwoMax-_axisTwoMin)/(100.*(nbinsTwo-1))))
       throw cet::exception("BADCONFIG") << "BField mapping axis values not steppable with step size given: "
 					<< _axisTwoMin << " < axisTwoValues < " << _axisTwoMax << ", "
 					<< "step size = " << _mapBinSize;
 
     //if a null map, plot the default field from the manager
     const std::string name = (map) ? map->getKey() : "default";
-
     if(_hMap[name]) return; //already filled the maps in a previous event
 
+    // if dumping, create the file
+    std::fstream fs;
+    if(_dump){
+      string dumpfile= name+_dumpname+".csv";
+      fs.open (dumpfile.c_str(), fstream::out);
+      if(fs.is_open()){
+	fs << "# Dump of " << name << endl;
+	fs <<  "#  x   ,  y   ,   z   ,  Bx   ,  By   ,  Bz" << endl;
+      }
```

Comment:
Adding an else clause that throws an exception on failure to open the file would be prudent.

Reply:
OK